### DISCUSSION TITLE
Frigate to GPIO

### DISCUSSION THREAD

**Discussion by DuncanCT** [blakeblackshear/frigate#7101] [Ask A Question]
I'm running Frigate on a Dell OptiPlex Micro. I'm looking for a way to trigger an input on a Paradox Alarm panel. The best option I can think of at the moment is using USB to GPIO, something like this - [https://www.adafruit.com/product/2264]. Assuming that, you would also need something to convert MQTT to USB to GPIO. Is there an easier way or does anyone have a better idea?

**Comment by NateMeyer** [comment]
I would look at an ESP32 or ESP8266 devkit.  They can connect to wifi and monitor the MQTT messages directly and provide GPIO.  The ESPHome project should make it simple enough to set up any actions.

**Comment by DuncanCT** [comment]
Thanks Nate, but I've got Home Assistant running so I could push there and get ESPHome to trigger the zone but that seems like an unnecessary round trip. Looking to see if there is a simple and robust solution.

**Reply by hagak** [reply]
The extra round trip will be in the 1-2ms range most likely, maybe 4ms since will be wifi.

**Reply by DuncanCT** [reply]
Sorry, round trip probably not the correct terminology.

Not worried about the extra ms, but more from a reliability standpoint. Including extra sevices will just add potential problems. Instead of just Frigate, you then have to rely on MQTT, Home Assistant, ESPHome and Wifi.

I assume it's not possilble but it would be nice if Frigate had suport for USB/GPIO output.

**Reply by hagak** [reply]
Seems a bit out of scope for Frigate.  You can eliminate the need for MQTT by using ESPhome with HA.  HA should be the hub for your controls, ESPHome is really just your micro-controller there is not "service" that needs to be function just the ESP device.  I have about 60 of them and they are all rock solid.

So then the reliablity is on your WiFi and well that can be handled with multiple APs so that if one goes offline your devices will connect to the others.

**Reply by DuncanCT** [reply]
What if you just want to deloy Frigate on a remote site without HA/MQTT, wifi and whatever ESP device needed? Assume you're just trying to tell me it's not possible?

**Comment by scooper1** [comment]
use a shelly 1  relay contact to trigger an alarm?
